homeassistant/components/homekit/type_sensors.py

```diff
@@ -75,3 +75,46 @@ def update_state(self, entity_id=None, old_state=None, new_state=None):
             self.char_humidity.set_value(humidity, should_callback=False)
             _LOGGER.debug('%s: Percent set to %d%%',
                           self.entity_id, humidity)
+
+
+@TYPES.register('BinarySensor')
+class BinarySensor(HomeAccessory):
+    """Generate a BinarySensor accessory as binary sensor."""
+
+    def __init__(self, hass, state, **kwargs):
+        """Initialize a BinarySensor accessory object."""
+        entity_id = state.entity_id
+        super().__init__(state.name, entity_id, CATEGORY_SENSOR, **kwargs)
+
+        self.hass = hass
+        self.entity_id = entity_id
+
+        service_map = {
+            'gas': ('CarbonMonoxideSensor', 'CarbonMonoxideDetected'),
+            'co2': ('CarbonDioxideSensor', 'CarbonDioxideDetected'),
+            'occupancy': ('OccupancySensor', 'OccupancyDetected'),
+            'opening': ('ContactSensor', 'ContactSensorState'),
+            'motion': ('MotionSensor', 'MotionDetected'),
+            'moisture': ('LeakSensor', 'LeakDetected'),
+            'smoke': ('SmokeSensor', 'SmokeDetected')}
+
+        device_class_key = 'homekit_device_class' \
+            if ('homekit_device_class' in state.attributes) \
+            else 'device_class'
```

Comment:
Currently there is no `homekit_device_class` attribute. For this PR we should concentrate just on the `device_class`.

Reply:
homekit_device_class it just like homebridge_outlet_type, etc in HomeBridge/homegbridge+homeassistant. It's preserved for user to assign this param in customization.yaml manually. Because sometimes there is no device_class in binary_sensor's attributes, and user can assign it to gas co/co2, or any other binary_sensor.

There will be more homekit_xxxx key for CO2/Light/AirQuality sensor to get correct sensor type instead of guessing by unit or entity id.  I'll document it in PR comment.

Reply:
We shouldn't write those in the state machine. I would prefer to leave it at `device_class` only for the moment. If you really want to implement this, use the `entity_config` dictionary from the `homekit` component. A good example is the `alarm_code` for `type_security_systems.py`.

Reply:
Remove it at this time.